api/v1beta2/conversion_hub.go

```diff
@@ -0,0 +1,348 @@
+		proxySettings := make([]ProxySettings, 0, len(owner.ProxyOperations))
+
+		for _, proxyOp := range owner.ProxyOperations {
+			ops := make([]ProxyOperation, 0, len(proxyOp.Operations))
+
+			for _, op := range proxyOp.Operations {
+				ops = append(ops, ProxyOperation(op))
+			}
+
+			proxySettings = append(proxySettings, ProxySettings{
+				Kind:       ProxyServiceKind(proxyOp.Kind),
+				Operations: ops,
+			})
+		}
+
+		in.Spec.Owners = append(in.Spec.Owners, OwnerSpec{
+			Kind:            OwnerKind(owner.Kind),
+			Name:            owner.Name,
+			ClusterRoles:    owner.GetRoles(*src, index),
+			ProxyOperations: proxySettings,
+		})
+	}
+
+	if nsOpts := src.Spec.NamespaceOptions; nsOpts != nil {
+		in.Spec.NamespaceOptions = &NamespaceOptions{}
+
+		in.Spec.NamespaceOptions.Quota = src.Spec.NamespaceOptions.Quota
+
+		if metadata := nsOpts.AdditionalMetadata; metadata != nil {
+			in.Spec.NamespaceOptions.AdditionalMetadata = metadata
+		}
+
+		if value, found := annotations[capsulev1beta1.ForbiddenNamespaceLabelsAnnotation]; found {
+			in.Spec.NamespaceOptions.ForbiddenLabels.Exact = strings.Split(value, ",")
+
+			delete(annotations, capsulev1beta1.ForbiddenNamespaceLabelsAnnotation)
+		}
+
+		if value, found := annotations[capsulev1beta1.ForbiddenNamespaceLabelsRegexpAnnotation]; found {
+			in.Spec.NamespaceOptions.ForbiddenLabels.Regex = value
+
+			delete(annotations, capsulev1beta1.ForbiddenNamespaceLabelsRegexpAnnotation)
+		}
+
+		if value, found := annotations[capsulev1beta1.ForbiddenNamespaceAnnotationsAnnotation]; found {
+			in.Spec.NamespaceOptions.ForbiddenAnnotations.Exact = strings.Split(value, ",")
+
+			delete(annotations, capsulev1beta1.ForbiddenNamespaceAnnotationsAnnotation)
+		}
+
+		if value, found := annotations[capsulev1beta1.ForbiddenNamespaceAnnotationsRegexpAnnotation]; found {
+			in.Spec.NamespaceOptions.ForbiddenAnnotations.Regex = value
+
+			delete(annotations, capsulev1beta1.ForbiddenNamespaceAnnotationsRegexpAnnotation)
+		}
+	}
+
+	if svcOpts := src.Spec.ServiceOptions; svcOpts != nil {
+		in.Spec.ServiceOptions = &api.ServiceOptions{}
+
+		if metadata := svcOpts.AdditionalMetadata; metadata != nil {
+			in.Spec.ServiceOptions.AdditionalMetadata = metadata
+		}
+
+		if allowed := svcOpts.AllowedServices; allowed != nil {
+			in.Spec.ServiceOptions.AllowedServices = allowed
+		}
+
+		if externalIPs := svcOpts.ExternalServiceIPs; externalIPs != nil {
+			in.Spec.ServiceOptions.ExternalServiceIPs = externalIPs
+		}
+	}
+
+	if sc := src.Spec.StorageClasses; sc != nil {
+		in.Spec.StorageClasses = &api.AllowedListSpec{
+			Exact: sc.Exact,
+			Regex: sc.Regex,
+		}
+	}
+
+	if scope := src.Spec.IngressOptions.HostnameCollisionScope; len(scope) > 0 {
+		in.Spec.IngressOptions.HostnameCollisionScope = scope
+	}
+
+	v, found := annotations[capsulev1beta1.DenyWildcard]
+	if found {
+		value, err := strconv.ParseBool(v)
+		if err == nil {
+			in.Spec.IngressOptions.AllowWildcardHostnames = value
```

Comment:
Thanks for pointing this out, always get puzzled with boolean values!

JFI opted for an __allow__ approach rather than __deny__ since from the end user perspective is much simpler to understand.